### PR TITLE
OGM-679 EmbeddableMappingTest ported to CouchDB

### PR DESCRIPTION
**Pull request by mrizzi**
EmbeddableMappingTest ported from MongoDB to CouchDB enhanced with a further test to verify the behaviour when setting to null an @Embeddable object

### CODE REVIEW

**Comment by gunnarmorling** [comment]
ok to test

File: couchdb/src/test/java/org/hibernate/ogm/datastore/couchdb/test/embeddable/EmbeddableMappingTest.java [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+					"'login' : 'emmanuel', " +
+					"'homeAddress' : {" +
+						"'city' : 'Paris', " +
+						"'country' : 'France', " +
+						"'street1' : '1 avenue des Champs Elysees'," +
+						"'type' : {" +
+							"'name' : 'main'" +
+						"}" +
+					"}, " +
+					"'postal_code' : '75007', " +
```
Comment (gunnarmorling) [inline couchdb/src/test/java/org/hibernate/ogm/datastore/couchdb/test/embeddable/EmbeddableMappingTest.java]: Could you add a TODO comment pointing to OGM-893? That'll avoid potential confusion as this is not really what one would expect.

File: couchdb/src/test/java/org/hibernate/ogm/datastore/couchdb/test/embeddable/EmbeddableMappingTest.java [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+				session.getSessionFactory(),
+				// collection
+				"Account",
+				// query
+				"Account:login_:emmanuel_",
+				// expected
+				"{ " +
+						"'login' : 'emmanuel', " +
+						"'homeAddress' : {" +
+						"'city' : 'Paris', " +
```
Comment (gunnarmorling) [inline couchdb/src/test/java/org/hibernate/ogm/datastore/couchdb/test/embeddable/EmbeddableMappingTest.java]: The indentation looks odd.

File: couchdb/src/test/java/org/hibernate/ogm/datastore/couchdb/test/embeddable/EmbeddableMappingTest.java [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+		// Then
+		assertDbObject(
+				session.getSessionFactory(),
+				// collection
+				"Account",
+				// query
+				"Account:login_:emmanuel_",
+				// expected
+				"{ " +
+						"'login' : 'emmanuel', " +
```
Comment (gunnarmorling) [inline couchdb/src/test/java/org/hibernate/ogm/datastore/couchdb/test/embeddable/EmbeddableMappingTest.java]: Same here; We try to indent JSON strings properly to ease reading.

**Comment by gunnarmorling** [comment]
@mrizzi Looks good overall, just added some minor remarks. Could you also squash the three commits into one so to keep the history mean and lean. Then it's good to go :) Thanks!

**Comment by Sanne** [comment]
add to whitelist

**Comment by gunnarmorling** [comment]
Rebased and applied. Many thanks, @mrizzi! Sorry for the late response btw., I didn't notice you had send an update for this pull request. Just add a short comment when doing so to be on the safe side :)

**Pull request closed without merging**